scout/run.py

```diff
@@ -5097,6 +5784,1366 @@ def out_break_walk_subtr(self, orig_dict, base_val, eff_val, focus_yrs):
+                    self.apply_onsite_reduce(
+                        brk_dat_eff, brk_dat_eff_capt, brk_dat_eff_capt_env, brk_dat_base,
+                        brk_dat_save, mast_dat_eff, mast_dat_base, mast_dat_save, brk_dat_cdbps_eff,
+                        brk_dat_cdbps_base, brk_dat_cdbps_save, mast_dat_cdbps_eff,
+                        mast_dat_cdbps_base, mast_dat_cdbps_save, reg, bldg, apply_yrs,
+                        onsite_times_apply_fracs, rel_elec_eff[var], prior_yr_rmv, var,
+                        cdbps_regs, cdbps_bldgs, cdbps_eus, focus_yrs, reg, bldg, vint, meas_fuel,
+                        meas_eus, res_focus)
+                # Apply any additional energy reduction requirements, leveraging data shorthands
+                # above. Note that stock remains the same for code reductions, such that per unit
+                # energy/carb/cost will be reduced
+                elif var != "stock":
+                    self.apply_energy_reduce(
+                        brk_dat_eff, brk_dat_eff_capt, brk_dat_eff_capt_env, brk_dat_base,
+                        brk_dat_save, mast_dat_eff, mast_dat_base, mast_dat_save, brk_dat_cdbps_eff,
+                        brk_dat_cdbps_base, brk_dat_cdbps_save, mast_dat_cdbps_eff,
+                        mast_dat_cdbps_base, mast_dat_cdbps_save, reg, bldg, apply_yrs,
+                        add_energy_times_apply_fracs, prior_yr_rmv, cdbps_regs, cdbps_bldgs,
+                        cdbps_eus, focus_yrs)
+
+        return
+
+    def apply_onsite_reduce(
+            self,  brk_dat_eff, brk_dat_eff_capt, brk_dat_eff_capt_env, brk_dat_base, brk_dat_save,
+            mast_dat_eff, mast_dat_base, mast_dat_save, brk_dat_cdbps_eff, brk_dat_cdbps_base,
+            brk_dat_cdbps_save, mast_dat_cdbps_eff, mast_dat_cdbps_base, mast_dat_cdbps_save,
+            reg, bldg, apply_yrs, onsite_times_apply_fracs, rel_elec_eff,
+            prior_yr_rmv, var, cdbps_regs, cdbps_bldgs, cdbps_eus, focus_yrs, reg_in, bldg_in,
+            vint_in, meas_fuel, meas_eus, res_focus):
+        """Apply onsite emissions reductions required via code/BPS.
+
+        Args:
+            brk_dat_eff (dict): Efficient stock/energy/carbon/ecost breakouts for indiv. measure.
+            brk_dat_eff_capt (dict): Efficient-captured energy breakouts for indiv. measure.
+            brk_dat_eff_capt_env (dict): Efficient-captured-envelope breakouts for indiv. measure.
+            brk_dat_base (dict): Baseline stock/energy/carbon/ecost breakouts for indiv. measure.
+            brk_dat_save (dict): Energy/carbon/ecost savings breakouts for indiv. measure.
+            mast_dat_eff (dict): Master efficient data for indiv. measure.
+            mast_dat_base (dict): Master baseline data for indiv. measure.
+            mast_dat_save (dict): Master savings data for indiv. measure.
+            brk_dat_cdbps_eff (dict): Eff. stock/energy/carbon/cost breakouts for codes/BPS measure.
```

Comment:
There are so many similar variable names that are hard to track, I wonder if there is a better way to pass this data. Sorry no actual suggestion at the moment.